src/main/java/seedu/address/logic/parser/AddBuyerCommandParser.java

```diff
@@ -41,12 +43,12 @@ public AddBuyerCommand parse(String args) throws ParseException {
         }
 
         argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_INFO);
-        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
-        Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
-        BuyHouseInfo buyHouseInfo = ParserUtil.parseBuyHouseInfo(argMultimap.getValue(PREFIX_INFO).get());
-        Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
+        Name name = ParserUtil.parseName(warn, argMultimap.getValue(PREFIX_NAME).get());
+        Phone phone = ParserUtil.parsePhone(warn, argMultimap.getValue(PREFIX_PHONE).get());
+        Email email = ParserUtil.parseEmail(warn, argMultimap.getValue(PREFIX_EMAIL).get());
+        Address address = ParserUtil.parseAddress(warn, argMultimap.getValue(PREFIX_ADDRESS).get());
+        BuyHouseInfo buyHouseInfo = ParserUtil.parseBuyHouseInfo(warn, argMultimap.getValue(PREFIX_INFO).get());
+        Set<Tag> tagList = ParserUtil.parseTags(warn, argMultimap.getAllValues(PREFIX_TAG));
```

Comment:
Nice idea of passing around a warning object to be able to add warning messages at every step of the parsing!